presto-main/src/main/java/com/facebook/presto/operator/aggregation/BlackholeAggregation.java

```diff
@@ -0,0 +1,137 @@
+import static com.facebook.presto.common.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.common.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.operator.aggregation.AggregationUtils.generateAggregationName;
+import static com.facebook.presto.spi.function.Signature.comparableTypeParameter;
+import static com.facebook.presto.spi.function.aggregation.AggregationMetadata.ParameterMetadata;
+import static com.facebook.presto.spi.function.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.BLOCK_INDEX;
+import static com.facebook.presto.spi.function.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.NULLABLE_BLOCK_INPUT_CHANNEL;
+import static com.facebook.presto.spi.function.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.STATE;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.airlift.slice.Slices.utf8Slice;
+
+/**
+ * Aggregating function that does nothing on the input data and always returns
+ * the "OK" varchar value.
+ *
+ * It works similar to the CHECKSUM function, but it does not compute the hash to
+ * minimize resource usage. It can be used when you need to scan all data in a table.
+ */
+public class BlackholeAggregation
+        extends SqlAggregationFunction
+{
+    public static final BlackholeAggregation BLACKHOLE_AGGREGATION = new BlackholeAggregation();
+    private static final String NAME = "blackhole";
+    private static final Slice OUTPUT = utf8Slice("OK");
+    private static final MethodHandle OUTPUT_FUNCTION = methodHandle(BlackholeAggregation.class, "output", NullableLongState.class, BlockBuilder.class);
+    private static final MethodHandle INPUT_FUNCTION = methodHandle(BlackholeAggregation.class, "input", Type.class, NullableLongState.class, Block.class, int.class);
+    private static final MethodHandle COMBINE_FUNCTION = methodHandle(BlackholeAggregation.class, "combine", NullableLongState.class, NullableLongState.class);
+
+    public BlackholeAggregation()
+    {
+        super(NAME,
+                ImmutableList.of(comparableTypeParameter("T")),
+                ImmutableList.of(),
+                parseTypeSignature(StandardTypes.VARBINARY),
+                ImmutableList.of(parseTypeSignature("T")));
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Consumes given values as a blackhole.";
+    }
+
+    @Override
+    public BuiltInAggregationFunctionImplementation specialize(BoundVariables boundVariables, int arity, FunctionAndTypeManager functionAndTypeManager)
+    {
+        Type valueType = boundVariables.getTypeVariable("T");
+        return generateAggregation(valueType);
+    }
+
+    private static BuiltInAggregationFunctionImplementation generateAggregation(Type type)
+    {
+        DynamicClassLoader classLoader = new DynamicClassLoader(BlackholeAggregation.class.getClassLoader());
```

Comment:
Is there an possibility to share implementation with ChecksumAggregationFunction?